src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -298,74 +298,174 @@ public void StartRecording()
 
         }
 
-        private KeyFrameType[] GetKeyFrameType(List<RecordedGameObjectState> priorState, List<RecordedGameObjectState> currentState, string pixelHash)
+        /**
+         * <summary>Modifies the lists with removal leaving only the unique entries in each list.  Assumes lists have unique entries as though created from a hashset</summary>
+         */
+        private bool IntListsMatch(List<int> priorScenes, List<int> currentScenes)
```

Comment:
I didn't want to add this comment since it really doesn't matter and I don't want to clog up the PR, but I had fun pondering if this approach or a sort + compare would be faster... with some constants included, sorting would be 2(nlogn + n) while this approach would potentially be n^2, but I bet with the `break;` and size of these lists, the practical analysis isn't as straightforward... 

Actually, an even faster approach, since the lists have unique entries, it might be even faster to just convert each list to a set, and then hash the sets (you'd have to implement that yourself though and that math is probably kinda weird), and compare the hashes... which would be O(n) I think... 

Once again, completely ignore this probably, I just miss doing this sort of algorithmic thinking

Reply:
There is also https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.setequals?view=net-8.0

Reply:
Tried and evaluated all these suggestions and more in many places in the code, in practice, given the relatively small sizes of our lists (normally in the 100s or 1000s, and < 10000) entries, the looping iteration is both faster and avoids lots of GC allocations that hashing does in all cases.

In this specific case the list is always less than 10 elements, so anything other than just checking is 1 to 2 orders of magnitude longer time in practice and incurs GC allocations.

Also.. on sorting .. I had the same thought that some of these n^2 cases.. sorting it first would be faster as I could quick search, but in practice the math comparisons and allocations required to do the sorting still took more time than just iterating for our data set sizes.

Reply:
got it, makes perfect sense! Thanks for entertaining my comment XD